### PR TITLE
feat(cli): add status and reset commands

### LINKED ISSUE #485: [FEATURE] Add command to drop locally pending measurements
## Feature Description

When I have done some test measurements that I do not intend to push, I want a command to see pending measurements like `git status` and a command like `git reset` to drop pending measurements.

### PR DESCRIPTION
**Pull request by kaihowl**
## Summary

Implements issue #485 by adding two new commands to manage locally pending measurements:

- **`git perf status`**: View pending measurements that haven't been pushed to the remote repository
- **`git perf reset`**: Discard all locally pending measurements

Additionally, aligns the output formatting of status and reset commands for consistency and readability, while preserving concurrency-safe write-ref handling to prevent race conditions during reset/push operations.

## Implementation Plan

This implementation follows the detailed plan from issue #485, broken down into 6 phases:

### Phase 1: Status Command - Basic Implementation
- ✅ Add `Status` command to `Commands` enum in `cli_types/src/lib.rs`
- ✅ Create `git_perf/src/status.rs` with core status logic
- ✅ Wire up command handler in `git_perf/src/cli.rs`
- ✅ Create consolidated read branch from write refs only (exclude remote)
- ✅ Use `walk_commits()` to gather measurement data

### Phase 2: Status Command - Enhanced Features
- ✅ Implement `--detailed` flag for per-commit breakdown
- ✅ Improve output formatting with commit/measurement counts
- ✅ List unique measurement names
- ✅ Add helpful hints for next steps
- ✅ Handle empty state with helpful message

### Phase 3: Reset Command - Core Implementation
- ✅ Add `Reset` command to `Commands` enum
- ✅ Create `git_perf/src/reset.rs` with reset logic
- ✅ Implement `--dry-run` flag to preview changes
- ✅ Implement `--force` flag to skip confirmation
- ✅ **Concurrency-safe implementation**: Create new write ref before deleting old ones (per amendment)
- ✅ Interactive confirmation prompt by default
- ✅ Safety: Only delete write refs, never touch published measurements

### Phase 4: Documentation Generation
- ✅ Add comprehensive doc comments to CLI definitions
- ✅ Include examples for both commands
- ✅ Explain pending vs published measurements
- ✅ Add safety warnings for reset
- ✅ Run `./scripts/generate-manpages.sh` to regenerate docs
- ✅ Generated `man/man1/git-perf-status.1` and `man/man1/git-perf-reset.1`
- ✅ Updated `docs/manpage.md`

### Phase 5: Integration Testing
- ✅ Created `test/test_status.sh` with comprehensive status tests
- ✅ Created `test/test_reset.sh` with reset tests including dry-run and force modes
- ✅ Tests cover edge cases: no pending measurements, multiple measurements, after push
- ✅ All tests pass successfully

### Phase 6: Code Quality & Polish
- ✅ Run `cargo fmt` to format all code
- ✅ Run `cargo clippy` and fix warnings
- ✅ Address visibility issues (make internal functions private)
- ✅ Fix usize::MAX overflow issue (use 1_000_000 limit)
- ✅ Remove unnecessary borrows
- ✅ Collapse nested if statements
- ✅ Proper error handling with Result types

### Status Command Flow
```
1. User runs: git perf status
2. create_consolidated_pending_read_branch()
   └─> Get all write refs: refs/notes/perf-v3-write-*
   └─> Consolidate into temp read ref (NO remote branch)
3. walk_commits() to iterate all commits with notes
4. deserialize() each note to get measurements
5. Count commits, collect unique measurement names
6. display_status() - format and print output
```

### Reset Command Flow (Concurrency-Safe)
```
1. User runs: git perf reset
2. plan_reset()
   └─> get_refs() to find all write refs
   └─> count_all_pending_measurements() for display
3. display_reset_plan() - show what will be deleted
4. confirm_reset() - interactive prompt (unless --force or --dry-run)
5. execute_reset()
   └─> Create new write ref: refs/notes/perf-v3-write-{unique_id}
       (This becomes the "canonical" empty state)
   └─> For each OTHER write ref: remove_reference(ref_name)
       (Reset all pre-existing write refs)
6. Print success message
```

### Status Command
Shows count of commits with pending measurements, lists unique measurement names, and supports `--detailed` flag for per-commit breakdown. Only examines local write refs, not the remote branch.

### Reset Command  
Creates new write ref before deleting old ones (concurrency-safe per amendment). Supports `--dry-run` to preview without making changes and `--force` to skip confirmation prompt. Only affects local write refs, never remote measurements.

### Write-ref Isolation Refactor
- Introduced `create_new_write_ref` in `git_perf/src/git/git_interop.rs` to atomically create a fresh write ref before old ones are deleted, enabling safe concurrent operations
- Added `create_consolidated_pending_read_branch` to build a temporary read branch from pending writes only (excluding remote)
- Added helpers: `get_write_refs` and `delete_reference` to support safer manipulation of write refs
- Reset flow now creates the new write ref first, then plans and deletes old write refs, ensuring concurrent writes aren't inadvertently removed
- Status flow uses consolidated pending reads to provide accurate local-only status reporting

### Internal Module Changes
- Exposed necessary components in the git module for tests and CLI integration
- Made internal functions private to avoid visibility lint errors

### Public API and Behavior Notes
- The reset path now guarantees a fresh write ref exists before deletions, preventing race conditions with ongoing writes
- The status path continues to report on local-only data by using a consolidated view of pending write refs

## Safety Guarantees

1. **Never affect published measurements**:
   - Only operate on write refs (`refs/notes/perf-v3-write-*`)
   - Never touch `refs/notes/perf-v3` (published branch)
   - Never interact with remote repository

2. **Confirmation and visibility**:
   - Show what will be deleted before deletion
   - Require explicit confirmation (unless `--force`)
   - `--dry-run` for safe preview

3. **Concurrency-safe**:
   - Creates new write ref before deleting others
   - Each process maintains isolation via separate write refs
   - No locks needed (git refs are atomic)

## Test Plan

Added integration tests:
- `test/test_status.sh` - Tests status command with various scenarios
- `test/test_reset.sh` - Tests reset command including dry-run and force modes

All tests pass:
```bash
./test/test_status.sh  # ✓ All status tests passed
./test/test_reset.sh   # ✓ All reset tests passed
```

Build and linting:
- ✅ `cargo build` - no errors
- ✅ `cargo clippy` - all warnings addressed
- ✅ `cargo fmt` - code formatted
- ✅ Documentation generated with `./scripts/generate-manpages.sh`

## Documentation

- CLI help text with examples in `cli_types/src/lib.rs`
- Generated manpages: `man/man1/git-perf-status.1`, `man/man1/git-perf-reset.1`
- Documentation generated in `docs/manpage.md`
- Implementation follows design in `docs/plans/status-and-reset-commands.md`

## Related

Closes #485

🤖 Generated with [Claude Code](https://claude.com/claude-code)

📎 **Task**: https://www.terragonlabs.com/task/26d99a75-53a6-46a4-8855-79b42761836e

📎 **Task**: https://www.terragonlabs.com/task/68facb6c-e7d6-4e41-bcf2-77a0e1ddf122

📎 **Task**: https://www.terragonlabs.com/task/883724dd-66bf-4c53-97ce-37bb8b416fd0

📎 **Task**: https://www.terragonlabs.com/task/38ce0a72-cc05-40ca-b709-6275b5349297

### CODE REVIEW

File: git_perf/src/reset.rs [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        commit_count,
+    })
+}
+
+/// Count all pending measurements
+fn count_all_pending_measurements() -> Result<(usize, usize)> {
+    // Create consolidated pending branch using git module function
+    let _guard = create_consolidated_pending_read_branch()?;
+    // Use a large but reasonable number instead of usize::MAX to avoid overflow issues
+    let commits = walk_commits(1_000_000)?;
```
Comment (kaihowl) [inline git_perf/src/reset.rs]: magic number

File: git_perf/src/status.rs [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        let measurements = deserialize(&note_text);
+
+        if measurements.is_empty() {
+            continue;
+        }
+
+        commit_count += 1;
+
+        // Collect unique measurement names
+        let commit_measurement_names: Vec<String> =
```
[the remark this answers is not in the document]
Comment (kaihowl) [reply git_perf/src/status.rs]: This is fine. Check the usage below.

File: git_perf/src/status.rs [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+    // Efficiently get commits that have notes in the pending branch
+    // These are all commits with pending (unpushed) measurements
+    let pending_commits = get_commits_with_notes(pending_ref)?;
+
+    let mut commit_count = 0;
+    let mut all_measurement_names = HashSet::new();
+    let mut per_commit = if detailed { Some(Vec::new()) } else { None };
+
+    for commit_sha in &pending_commits {
+        let note_lines = get_notes_for_commit(pending_ref, commit_sha)?;
```
Comment (kaihowl) [inline git_perf/src/status.rs]: This is inefficient. Instead of creating a new process for each commit, we should create one process to get all commits with a note on the temp read branch and then pipe that to the process for git show or even reuse the git log command.
